examples/html-wagmi/src/main.js

```diff
@@ -21,7 +21,13 @@ const wagmiConfig = defaultWagmiConfig({
 })
 
 // 3. Create modal
-const modal = createWeb3Modal({ wagmiConfig, projectId, chains, themeMode: 'light' })
+const modal = createWeb3Modal({
+  wagmiConfig,
+  projectId,
+  chains,
+  themeMode: 'light',
+  reconnect: true
```

Comment:
Similar to above, should reconnect be something we set internally and don't expose to user? Or perhaps this is a new flag we set to true by default and allow user to override? Thinking here is, how do we make it as simple and quick for user to get going without the need for all these extra options?

If they need to be enabled in ssr env and it has to be done manually, maybe we can combine it all under one flag?

Reply:
I feel setting this to true by default is better. 

Regarding merging the `ssr` flag, I'm pretty sure this would be required so merging would make sense, but tagging @glitch-txs jic

Reply:
this is not related to SSR but rather revalidation of the storage data on page load, this needs to be set manually by calling `reconnect(config)` function if you're using `@wagmi/core`, for those using React this is already handled internally (it's true by default).
This helps devs that use other frameworks than React to make revalidation more intuitive, for those using SSR should get this flag set to `false` and handle the `reconnect` function themselves. e.g. revalidating on the app "`onMount`" or similar.

Wagmi uses this to optimistically hydrate the storage with cookies and after the app is mounted reconnect is called to avoid hydration mismatches.

Reply:
> Similar to above, should reconnect be something we set internally and don't expose to user?

To clarify the above configuration, we added `...wagmiConfig` in defaultWagmiConfig so devs can add any extra Wagmi config's option they want, this is why you could add `ssr` or any other config

```ts
export function defaultWagmiConfig({
  projectId,
  chains,
  metadata,
  enableInjected,
  enableCoinbase,
  enableWalletConnect,
  enableEmail,
  enableEIP6963,
  ...wagmiConfig // <-- here
}: ConfigOptions): Config {
```

To add some context, reconnect is related to the revalidation of the storage data on page load (+ reconnecting), this needs to be set manually by calling `reconnect(config)` function if you're using `@wagmi/core`, for those using React this is already handled internally (it's true by default).

----------

Thinking a lot about this:

1.
> If they need to be enabled in ssr env and it has to be done manually, maybe we can combine it all under one flag?

If we do this and handle everything ourselves for those that are not using React then it might be a bit over-engineering imo, because this is something that Wagmi should handle instead.
There is for example a [Vue Wagmi library](https://github.com/unicape/use-wagmi) that might handle this internally as well, in this case devs won't be using tthis feature.

2. (rethinking this) It would be weird for a dev using SSR setting the `reconnect` flag to false, because they would still be able to reconnect if this is handled somewhere else.

I'm a bit confused now

should we leave this one and I can add the `reconnect` function manually in the documentation?

IMO let's leave this to wagmi and I can explain farther in the docs. (I just changed my mind xD)